external-builds/pytorch/smoke-tests/README.md

```diff
@@ -0,0 +1,28 @@
+# PyTorch Smoke Tests
+
+This repository contains a set of basic smoke tests for verifying fundamental PyTorch tensor operations. These tests ensure that basic matrix operations function correctly, but they are **not exhaustive PyTorch tests**.
+
+## Disclaimer
+
+⚠️ **These tests are not full PyTorch tests.** They are designed as quick checks for basic tensor operations and do not replace comprehensive testing of PyTorch functionalities.
```

Comment:
Should we also run slices of https://github.com/pytorch/pytorch/tree/main/test as recommended by https://rocm.docs.amd.com/projects/install-on-linux/en/latest/install/3rd-party/pytorch-install.html#testing-the-pytorch-installation ? Maybe we don't need our own local tests at all? Local tests would save us from cloning the full pytorch github repository though.

Reading through that page, we could also have `python3 -c 'import torch; print(torch.cuda.is_available())'` as a smoketest, if that isn't already available under the test/ folder.

Reply:
Having a subset of those tests would be great. The full Pytorch tests take a lot of time.

Reply:
Per our discussion, Scott and I decided to keep those smoke tests since beneficial to have a quick set of tests we can run. If it is needed, we can add more of those.

Reply:
Yeah it will be nice to have something simple that we directly control and can edit. As we test more we can clone the pytorch repo and run those tests.

We can do something like what the rocm docs recommend to run specific subsets:
```
PYTORCH_TEST_WITH_ROCM=1 python3 test/run_test.py --verbose \
--include test_nn test_torch test_cuda test_ops \
test_unary_ufuncs test_binary_ufuncs test_autograd
```

Reply:
As a subset, it sounds good to me.